### PR TITLE
Implement Rollbar for unhandled exceptions & Logging.Report

### PR DESCRIPTION
**Pull request by Tkael**
Fix for #284

### CODE REVIEW

File: Utilities/Logging.cs [hunk 1]
```diff
@@ -117,42 +120,157 @@ private static void log(string path, string method, string level, string data)
...
-                Thread thread = new Thread(() =>
+        public static void configureRollbarExceptionHandling(bool beta, string uniqueId)
+        {
+            var config = new RollbarConfig(rollbarWriteToken)
+            {
+                Environment = beta ? "development" : "production",
+                ScrubFields = new string[] // Scrub these fields from the reported data
                 {
-                    try
+                    "Commander", "apiKey", "commanderName"
```
[the remark this answers is not in the document]
Comment (Tkael) [reply Utilities/Logging.cs]: The EDSM API key. I didn't find a field to scrub for Frontier API credentials... we never really store them.

File: EDDI/Eddi.csproj [hunk 2]
```diff
@@ -0,0 +1,268 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (Tkael) [reply EDDI/Eddi.csproj]: Likewise. This wasn't part of the diff I was looking at either.

File: Utilities/Logging.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardbuckle) [reply Utilities/Logging.cs]: `EddiCompanionAppService.CompanionAppCredentials`. Pretty sure we're OK here and not at risk of posting anything we shouldn't.

File: EDDI/Eddi.csproj (hunk 2 above)
[the remark this answers is not in the document]
Comment (richardbuckle) [reply EDDI/Eddi.csproj]: 🤷‍♂️ looks like GitHub had a senior moment. I trust my local diffs.

File: Utilities/Logging.cs [hunk 3]
```diff
@@ -117,42 +120,178 @@ private static void log(string path, string method, string level, string data)
...
+            {
+                Exception exception = args.ExceptionObject as Exception;
 
-                string body = @"{""message"":""" + message + @""", ""version"":""" + Constants.EDDI_VERSION + @""", ""json"":" + data + @"}";
-                Thread thread = new Thread(() =>
+                Dictionary<string, object> trace = new Dictionary<string, object>();
+                if (exception.StackTrace != null)
                 {
-                    try
+                    trace.Add("StackTrace", exception.StackTrace != null ? exception.StackTrace : "StackTrace not available");
```
Comment (richardbuckle) [inline Utilities/Logging.cs]: Due to the outer test, the else condition will never fire here. I'll rework it with the ?? operator

**Pull request merged**